### ISSUE TITLE
PAT permissions issues for CD auto-commit for docs pages

### ISSUE THREAD

**Issue by aj-stein-nist** [usnistgov/OSCAL#1726]
### Describe the bug

The use of a GitHub PAT associated with the oscalbuilder, with only `repo:public_repo` permissions, is insufficient to complete the builds in GitHub Actions. On the final stage in different workflows, where updated model document and website content are committed back to the repo, the token lacks sufficient permissions and a `git clone` operation is denied.

Details as reported in a related issue can be found in https://github.com/usnistgov/OSCAL/issues/1698#issuecomment-1483622356.

### Who is the bug affecting

NIST OSCAL developer team wishing to release code, documentation, and website enhancements.

### What is affected by this bug

CI/CD, OSCAL Content, Documentation, Website

### How do we replicate this issue

1. Set up a developer fork of the repo.
2. Set up a PAT for the fork as explained in the screenshot in https://github.com/usnistgov/OSCAL/issues/1698#issuecomment-1483622356 and configure it for `username/OSCAL` fork.
3. Run the generate website workflow from your fork.
4. Review logs and see `git clone` failures on the generate website workflow, even without the presence of 429 errors.

### Expected behavior (i.e. solution)

The `git clone` operations are successfully completed without error, commits are made, and the the committed changes are pushed back to the repo so the website can be updated (this is how pages.nist.gov works).

**Comment by aj-stein-nist** [comment]
Now I am more confused, it seems that the `repo:public_repo` scope is sufficient on my fork, I guess I need to go back and determine why that is not the case on usnistgov/OSCAL, if at all?

**Comment by aj-stein-nist** [comment]
Now I am wondering if something about regenerating the token makes a difference, because lo and behold, even running it on the official repo, not a fork, is working without failing with simple `repo:public_repo` permissions, but I might have to make a minor change to the website to test this more thoroughly. Our opportunity may come in the form of merging #1662 later in this sprint as we come to a close, so I will test a little more and leave this under review until further notice.

Also, I will add a SOP page to the wiki section around debugging and troubleshooting this, as it seems very much necessary for others in this position moving forward.

**Comment by aj-stein-nist** [comment]
OK, [a workflow run on an unrelated branch on `developer` after merging a fresh PR](https://github.com/usnistgov/OSCAL/actions/runs/4567315037/jobs/8061179721) with the updated token failed once then worked. The updated token helped, but also the oscalbuilder account associated with the token used for committing needs the `Maintain` permissions on the usnistgov/OSCAL repo to bypass branch protections for direct commits to the protected branch. (It was already set up to be allowed to edit that branch, but it does not seem to be enough, see screenshot below.)

![image](https://user-images.githubusercontent.com/94922603/228923947-c3d21b33-350d-4be9-957c-038e283d884c.png)

**Comment by aj-stein-nist** [comment]
Well, that didn't go as planned, need to examine this further ... again, sigh.

**Comment by aj-stein-nist** [comment]
Previous debugging approaches and a prior solution didn't seem to change it anymore. I guess I will have to dig some more, but I am definitely opening a GH support ticket because the amount of effort I expended on this is now very excessive. I will test further more with the 429 header check feature branch of Chris but testing off my fork and a complete clone that is not fork connected does not afford the same granularity of branch protections, so I cannot completely test this with oscalbuilder on aj-stein-nist/OSCAL either (as opposed to my fork I used for a year or more now, relocated to aj-stein-nist/OSCAL-forked). Either way the subset of permissions on a personal fork, as opposed to a repo, is still problematic.

More to follow.
